### ISSUE TITLE
Increase disk space for systempool on privatek8s

### ISSUE THREAD

**Issue by smerle33** [jenkins-infra/helpdesk#3539]
### Service(s)

infra.ci.jenkins.io, release.ci.jenkins.io

### Summary

as per the alert `Available disk space is below 19.753% free for aks-systempool-36786399-vmss000000-publick8s` we need to increase the disk size from 30 to 40 or 50 as for linuxpool  (https://github.com/jenkins-infra/azure/commit/8a36feff1cfd53d34327e769bc3d05e8397178de) 

We foresee a service interruption of about 30mn as some services will have to restart (coredns, public-nginx-ingress, private-nginx-ingress, etc ...)
We plan this operation today 24 April 2023 during the afternoon (Paris time / in 3 to 4 hours)

we also plan a service restart for rss2twitter, ircbot, github-comment-ops as those 3 applications are running on this pool (migration to come, other issue to plan this https://github.com/jenkins-infra/helpdesk/issues/3540).

**Comment by dduportal** [comment]
As seen together, https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#os_disk_size_gb mentions that changing the value for the `default_node_pool` means re-creating the resource (e.g. the cluster).

Proposal: lets see if https://github.com/jenkins-infra/helpdesk/issues/3540 does not solve the "warning disk usage > 80%" messages

**Comment by dduportal** [comment]
Note: https://learn.microsoft.com/en-us/azure/aks/use-system-pools?tabs=azure-cli#delete-a-system-node-pool mentions that a 2nd system node pool is required in order to delete one. Tha explains the behavior in the Terraform.

Gotta check if https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#temporary_name_for_rotation would work

**Comment by dduportal** [comment]
I'm going to run this method: https://www.danielstechblog.io/how-to-change-the-node-size-of-the-default-node-pool-in-aks-without-downtime/ to update the default system node pool to 50 Gb.

TL;DR;
- [ ] Create a new temporary node pool in "System" mode
- [ ] Disable auto-scaling with the `az` cli on the current default node pool
- [ ] Drain workload out from default node pool (should be rescheduled on the new system node pool)
- [ ] Remove the original node pool (az CLI)
- [ ] Adjust terraform name for system node pool and refresh to check nothing was broken

**Comment by dduportal** [comment]
- Job Terraform Azure is enabled, ran with success and is green with no changes
- Azure Ui shows the system node pool is now using 50 Gb of (ephemeral) OS disk
- Datadog monitor is now fully green:

![Capture d’écran 2023-05-11 à 19 41 29](https://github.com/jenkins-infra/helpdesk/assets/1522731/bae1bb65-849d-42d9-8d58-c9fa25bb3b62)
